assets/js/components/ProjectEditor.js

```diff
@@ -62,6 +64,8 @@ export function ProjectEditor (projectDescriptionCredits, programId, textFields)
     }
   })
 
+  document.addEventListener('keyup', checkTextFields)
```

Comment:
Is there a more specific listener on each text fields? I thought there was a `lastSavedText` field that could be useful?

Reply:
The lastSavedText holds the last saved value for the text field so it is really only useful to check if the current text is the same as the last saved text.

I tried to see if there were any other handlers for the text fields that would trigger when the text field was changed, but I couldn't really find anything more straightforward than the 'keyup' event listener. It seems to work pretty well for what we want right now.

Reply:
have you tried this? https://stackoverflow.com/questions/6169353/html-text-input-onchange-event
i'm a bit concerned with the listener on the whole document, instead of the specific text fields

Reply:
Also we can add a small delay with debouncing so things don't react so often
https://stackoverflow.com/questions/27787768/debounce-function-in-jquery

Reply:
Updated it to a .change listener which is only called when the text field is changed and it is unselected so it does not get called as nearly as often